File: app/java/src/main/java/com/example/production/domain/model/subcontract/Consumption.java
package com.example.production.domain.model.subcontract;

import lombok.*;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;

/**
 * 消費データ
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Consumption {
    private Integer id;
    private String consumptionNumber;
    private String receivingNumber;
    private LocalDate consumptionDate;
    private String supplierCode;
    private String remarks;
    private LocalDateTime createdAt;
    private String createdBy;
    private LocalDateTime updatedAt;
    private String updatedBy;

    // 明細リスト（リレーション）
    private List<ConsumptionDetail> details;
}

File: app/csharp/src/ProductionManagement.Domain/Models/Subcontract/Consumption.cs
namespace ProductionManagement.Domain.Models.Subcontract;

/// <summary>
/// 消費データ
/// </summary>
public class Consumption
{
    public int Id { get; set; }
    public required string ConsumptionNumber { get; set; }
    public required string ReceivingNumber { get; set; }
    public required DateOnly ConsumptionDate { get; set; }
    public required string SupplierCode { get; set; }
    public string? Remarks { get; set; }
    public DateTime? CreatedAt { get; set; }
    public string? CreatedBy { get; set; }
    public DateTime? UpdatedAt { get; set; }
    public string? UpdatedBy { get; set; }

    /// <summary>
    /// 消費明細
    /// </summary>
    public IReadOnlyList<ConsumptionDetail> Details { get; set; } = [];
}
